crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -609,6 +618,24 @@ mod tests {
         let _ = unsafe { BlobVec::new(item_layout, Some(drop), 0) };
     }
 
+    #[test]
+    #[should_panic(expected = "ZST capacity overflow")]
+    fn blob_vec_zst_size_overflow() {
+        // SAFETY: no drop is correct drop for `()`.
+        let mut blob_vec = unsafe { BlobVec::new(Layout::new::<()>(), None, 0) };
+        // SAFETY: ZST, trivial drop, and ZST `BlobVec` has capacity `usize::MAX`.
```

Comment:
```suggestion
        // SAFETY: Because () is  a ZST type,, we can arbitrarily set the length and still be sound. It has a  trivial drop, and `BlobVec`s for ZSTs has capacity `usize::MAX`.
```